spec/compiler/semantic/const_spec.cr

```diff
@@ -366,6 +366,15 @@ describe "Semantic: const" do
       "can't declare constant dynamically"
   end
 
+  it "errors on dynamic constant assignment inside a block assigned to a constant (#2874)" do
+    assert_error %(
+      A = do
```

Comment:
```suggestion
      A = begin
```

Reply:
Ah wow... I've been AFK for a bit. Cheers @Sija